### PR TITLE
Fix Bulk Read

### PR DESCRIPTION
**Pull request by Maithem**
Removed Range Reads and added chucking to readAll.

### CODE REVIEW

**Comment by Maithem** [comment]
Needs more clean up, will do it tomorrow.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **8a2b7d8**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #884 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/Bgic7Pp7t3VzQn8t6W4cCLzDqo68nybJ)

File: runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java [hunk 1]
```diff
@@ -300,17 +303,25 @@ public void invalidateClientCache() {
...
      */
     public @Nonnull Map<Long, ILogData> cacheFetch(Iterable<Long> addresses) {
-        //turn the addresses into Set for now to satisfy signature requirement down the line
-        Set<Long> readAddresses = new TreeSet<>();
-        Iterator<Long> iterator = addresses.iterator();
-        while(iterator.hasNext()){
-            readAddresses.add(iterator.next());
+        Map<Long, ILogData> temp = new HashMap<>();
+
+        int batchSize = 20;
```
Comment (slfritchie) [inline runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java]: This constant seems like something that someone soon will want to make configurable/adjustable?

File: test/src/test/java/org/corfudb/integration/StreamIT.java [hunk 2]
```diff
@@ -1,68 +1,53 @@
...
 import java.util.UUID;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * A set integration tests that exercise the stream API.
  */
 
-public class StreamIT extends AbstractIT {
```
Comment (slfritchie) [inline test/src/test/java/org/corfudb/integration/StreamIT.java]: Hm, the surgery here appears to have changed the integration style test into a unit test.  I think it'd be better if this test had adaptations made within the IT style, then, if you wish, add a new unit test(s).

Also, there are two large blocks of code commented out.  If they aren't useful, then perhaps it's better to remove this entire test and/or convert to unit test style only?

Comment (Maithem) [inline test/src/test/java/org/corfudb/integration/StreamIT.java]: im discarding the test related changes, I just reused that test to run on a live system

File: infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java [hunk 3]
```diff
@@ -184,13 +184,10 @@ public void write(CorfuPayloadMsg<WriteRequest> msg, ChannelHandlerContext ctx,
     @ServerHandler(type = CorfuMsgType.READ_REQUEST, opTimer = metricsPrefix + "read")
     private void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IServerRouter r,
                       boolean isMetricsEnabled) {
-        log.trace("log read: {} {}", msg.getPayload().getStreamID() == null
-                        ? "global" : msg.getPayload().getStreamID(),
-                msg.getPayload().getRange());
+        log.trace("log read: {}", msg.getPayload().getSet());
```
Comment (no2chem) [inline infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java]: read: is fine, the logger will print the calling class

File: runtime/src/main/java/org/corfudb/protocols/wireprotocol/ReadRequest.java [hunk 4]
```diff
@@ -16,36 +18,24 @@
 @AllArgsConstructor
 public class ReadRequest implements ICorfuPayload<ReadRequest> {
 
-    final Range<Long> range;
-    @SuppressWarnings({"checkstyle:abbreviationaswordinname", "checkstyle:membername"})
-    final UUID streamID;
+    final Set<Long> set;
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/protocols/wireprotocol/ReadRequest.java]: Hm, let's not use set here - use a list. If the caller requests multiple addresses twice, we'll just return it, no need to enforce this.

File: runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java [hunk 5]
```diff
@@ -300,17 +303,25 @@ public void invalidateClientCache() {
...
+        Iterable<List<Long>> batches = Iterables.partition(addresses, batchSize);
+
+        for (List<Long> batch : batches) {
+            //turn the addresses into Set for now to satisfy signature requirement down the line
+            Set<Long> readAddresses = new TreeSet<>();
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java]: See above comment to keep the data structure a list down to the message

**Comment by Maithem** [comment]
@no2chem I will introduce a range read in another iteration. Let's phase this in one at a time, we dont have code that uses a ranged read.

**Comment by no2chem** [comment]
@Maithem well, the existing code uses a ranged read, so I would prefer if we preserved it...

**Comment by Maithem** [comment]
@no2chem I didn't say we shouldn't. @medhavidhawan recommended that we phase them in two different patches.

**Comment by no2chem** [comment]
@Maithem well, in this patch, create a new API (READ_MULTIPLE)? Instead of getting rid of the previous

**Comment by Maithem** [comment]
@no2chem So introduce a new api and use it while leaving the other API broken?

**Comment by no2chem** [comment]
@Maithem unless I'm mistaken, the other API works properly, it was just (1) used improperly (2) lacks an argument/bound check

**Comment by Maithem** [comment]
@no2chem The READ_REQUEST api, conflates two apis, read a single address and read range.

**Comment by no2chem** [comment]
@Maithem Ok, in a future PR, rename it READ_RANGE. But leave it intact, unless you can provide evidence to your claim that it is broken.

**Comment by Maithem** [comment]
Will open a new pr

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **b753944**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #884 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/MnFIfC5X5h22LnEMVClqj6rEykdNlWQ2)

**Pull request closed without merging**